src/providers/aws/mod.rs

```diff
@@ -49,29 +50,92 @@ pub struct AwsProvider {
 
 impl AwsProvider {
     pub fn try_new() -> Result<AwsProvider> {
-        let client = retry::Client::try_new()?.return_on_404(true);
+        let mut client = retry::Client::try_new()?.return_on_404(true);
+        let token = AwsProvider::fetch_imdsv2_token(client.clone());
```

Comment:
Here, you can now use your new method and drop the logic below:
```rust
pub fn try_new() -> Result<AwsProvider> {
  let client = ...
  AwsProvider::with_client(client)
}
```

Doing this also helps making sure that the test-path and the reaworld-path are as close as possible.